### ISSUE TITLE
Need CI - regtest mode

### ISSUE THREAD

**Issue by sander2** [BitVM/BitVM#126]
Currently there is no CI running the tests. This is probably mostly due to the tests relying on testnet. It would be great if the tests could run against a regtest node so that a CI can be set up to run all tests in prs. This would also help a lot in local development - running tests would become a lot less painful

**Comment by lucidLuckylee** [comment]
I agree. Currently the Repo is in a quite unpolished state as we have been gathering code snippets and trying things out. As soon as we have a complete groth16 verifier that is split over multiple chunks and can be executed on-chain I want to focus on cleaning up the repo.

**Comment by justin-elementlabs** [comment]
@sander2 @lucidLuckylee , we have updated one of our branches to support regtest: https://github.com/elementlabs42/BitVM/pull/2

Relevant scripts are here: https://github.com/elementlabs42/BitVM/pull/2/commits/cd37fc1669dd128c43c054175789716f694bf11d

This test should pass:
```Rust
cargo test -- --nocapture test_musig2_peg_in
```

We're working on fixing tests still and cleaning up our repo.
